javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,371 @@
+
+{
+  item: 'switch'
+  purchasePrice: 399
+}
+
+------------------------------------------------------------------------------------------------ */
+
+const addPurchases = (arr) => {
+  const addtogether = arr.reduce( (accumulator, val, idx) => {
+    return accumulator + val.purchasePrice;
+  }, 0);
+  return addtogether;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfElements that, given an array as input, uses reduce to count the number of elements in the array.
+
+Note: You may not use the array's built-in length property.
+------------------------------------------------------------------------------------------------ */
+
+const countNumberOfElements = (arr) => {
+  const numchr = arr.reduce( (accumulator, val, idx) => {
+    return idx+1;
+  }, 0);
+  return numchr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named returnNames that, given the Star Wars data, below, uses reduce to return an array containing the names of the characters.
+------------------------------------------------------------------------------------------------ */
+
+let starWarsData = [{
+  name: 'Luke Skywalker',
+  height: '172',
+  mass: '77',
+  hair_color: 'blond',
+  skin_color: 'fair',
+  eye_color: 'blue',
+  birth_year: '19BBY',
+  gender: 'male',
+},
+{
+  name: 'C-3PO',
+  height: '167',
+  mass: '75',
+  hair_color: 'n/a',
+  skin_color: 'gold',
+  eye_color: 'yellow',
+  birth_year: '112BBY',
+  gender: 'n/a'},
+{
+  name: 'R2-D2',
+  height: '96',
+  mass: '32',
+  hair_color: 'n/a',
+  skin_color: 'white, blue',
+  eye_color: 'red',
+  birth_year: '33BBY',
+  gender: 'n/a'
+},
+{
+  name: 'Darth Vader',
+  height: '202',
+  mass: '136',
+  hair_color: 'none',
+  skin_color: 'white',
+  eye_color: 'yellow',
+  birth_year: '41.9BBY',
+  gender: 'male'
+},
+{
+  name: 'Leia Organa',
+  height: '150',
+  mass: '49',
+  hair_color: 'brown',
+  skin_color: 'light',
+  eye_color: 'brown',
+  birth_year: '19BBY',
+  gender: 'female'
+}];
+
+const returnNames = (arr) => {
+  const chrname = arr.reduce( (acc, val, idx) => {
+    acc.push( val.name);
+    return acc;
+  }, []);
+  return chrname;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named reversedString that takes in a string and returns a string with the letters in reverse order.
+
+Note: You must use reduce for this challenge. You may not use the built-in .reverse() string method.
+------------------------------------------------------------------------------------------------ */
+
+const reversedString = (str) => {
+  return str.split('').reduce((r, c)=> c + r, '');
```

Comment:
Nice concise solution. I did notice this is exactly the same as the solution Tiara has. Which is ok, but if the two of you are doing the problems together or sharing answers, just make sure you mention it in your submission.